fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/PIDs.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright 2016 The University of North Carolina at Chapel Hill
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.unc.lib.dl.fcrepo4;
+
+import static edu.unc.lib.dl.fcrepo4.RepositoryPathConstants.HASHED_PATH_DEPTH;
+import static edu.unc.lib.dl.fcrepo4.RepositoryPathConstants.HASHED_PATH_SIZE;
+
+import java.net.URI;
+import java.util.regex.Matcher;
+
+import edu.unc.lib.dl.fedora.PID;
+
+/**
+ * Provides static methods for creating PID objects
+ * 
+ * @author bbpennel
+ *
+ */
+public class PIDs {
+
+	private static Repository repository;
+
+	/**
+	 * Get a PID object for the given URI.
+	 * 
+	 * @param uri
+	 * @return new PID object for the given URI
+	 */
+	public static PID get(URI uri) {
+		return get(uri.toString());
+	}
+
+	/**
+	 * Get a PID object for the given identifier or URI string. Should either be
+	 * a fully qualified repository URI or follow the syntax for an identifier,
+	 * such as: deposit/uuid:0411cf7e-9ac0-4ab0-8c24-ff367e8e77f6
+	 * 
+	 * @param value
+	 * @return new PID object for the given identifier or URI
+	 */
+	public static PID get(String value) {
+		String id;
+		String qualifier;
+		String componentPath;
+		String repositoryPath;
+
+		if (value.startsWith(repository.getFedoraBase())) {
+			// Given value was a fedora path. Remove the base and decompose
+			repositoryPath = value;
+			String path = value.substring(repository.getFedoraBase().length());
+
+			Matcher matcher = RepositoryPathConstants.repositoryPathPattern.matcher(path);
+			if (matcher.matches()) {
+				qualifier = matcher.group(1);
```

Comment:
The group numbers?  They are indexes referencing groups over in this regular expression pattern https://github.com/UNC-Libraries/Carolina-Digital-Repository/blob/1aa1a2d8cfaaa7bcfa698745463a3515ba5bc3b3/fcrepo-clients/src/main/java/edu/unc/lib/dl/fcrepo4/RepositoryPathConstants.java#L41 (the other regex after it is used below similarly).  

These particular indexes/values are not likely to be used elsewhere.  I wonder if that should be taken as a sign that the patterns should be in this class rather than the constants one, but I was trying to put the pieces used to understand paths together in one place (RepositoryPathConstants.java)